tests/api_test.py

```diff
@@ -947,6 +947,12 @@ def test_jvp_mismatched_arguments(self):
       TypeError,
       "primal and tangent arguments to jax.jvp do not match.",
       lambda: api.jvp(lambda x: -x, (np.float16(2),), (np.float32(4),)))
+    # If primals and tangents are not of the same shape then raise error
+    self.assertRaisesRegex(
+      ValueError,
+      "jvp called with inconsistent primal and tangent shapes",
+      lambda: api.jvp(lambda x: x+1, (np.random.randn(10,),), (np.random.randn(20,),))
+    )
```

Comment:
I would also add a test case when the primal and/or tangent are Python scalar.

Reply:
I have tried to address all of this changes in the latest commit.

Thanks,
Gaurav

Reply:
I do not actually see a test that uses a Python scalar for the primals and tangents. You can replace one of `np.float32(20)` with `20.`.

Reply:
Sorry, I am not very clear on this. Do you mean make a test case like so
```jax.jvp(fun, (10.,), (20.,))```

Thanks,
Gaurav Pathak

Reply:
I think I understand what you meant to say and have tried to add the test case.

Thanks,
Gaurav Pathak

Reply:
That is what I meant, thanks!